packages/nimble-components/src/utilities/models/array-to-tree.ts

```diff
@@ -0,0 +1,121 @@
+// Modified from https://github.com/philipstanislaus/performant-array-to-tree/blob/v1.11.0/src/arrayToTree.ts
```

Comment:
Since this file is super integrated now and table specific we can just move it to table/models